### ISSUE TITLE
Downstream Windows_certificate resource error after 4.3.0

### ISSUE THREAD

**Issue by ldliy1** [chef-cookbooks/windows#561]
### Cookbook version
4.3.0

### Chef-client version
14.2.0

### Platform Details
Windows 2012.R2

### Scenario:
Previously with 4.2.5 I was able to import pem file with resource defined below.  But after 4.3.0 change the resource seems to fail with error below

### Steps to Reproduce:
windows_certificate 'c:\\blah\\blah.pem' do
  store_name 'ROOT'
  action :create
end

### Expected Result:
4.2.5 imported pem file as expected

### Actual Result:
amazon-ebs: [2018-06-12T08:36:57+10:00] FATAL: SystemCallError: windows_certificate[c:\blah\blah.pem] (blah::certificates line 12) had an error: SystemCallError: The request is not supported. - Unable to add certificate.

**Comment by ldliy1** [comment]
Thanks for your feedback @piyushawasthi .  Unfortunately I wont be able to share the PEM file.  But I will try different export formats for the cert... If you are happy to share you PEM file, please do so.  I can try with your file to eliminate that out of the equation.   I will do some more digging in the coming days and post update.  Thanks again.

**Comment by zl4bv** [comment]
For me, it occurs with certificates in DER format and `.cer` extension. It has started happening with no other changes to how I install certificates.

**Comment by piyushawasthi** [comment]
Hi @ldliy1 you can match your PEM file with this (https://github.com/chef-cookbooks/windows/blob/master/test/cookbooks/test/files/GlobalSignRootCA.pem) also please make sure your PEM file always start with `-----BEGIN CERTIFICATE-----` and end with `-----END CERTIFICATE-----`.

**Comment by piyushawasthi** [comment]
@zl4bv will you please update your certificate extension to `.der` and try, after that if you are still facing any issue just let us know.

**Comment by amcappelli** [comment]
I get this error as well using version 4.3.2 and a PEM format with `.cer` extension, with 4.2.5 it works fine. I tried changing it to `.pem` and that did not help.

PS C:\Users\> gwmi win32_operatingsystem | % caption
Microsoft Windows Server 2016 Datacenter
PS C:\Users\> chef-client -v
Chef: 13.8.5

**Comment by amcappelli** [comment]
my cert is `sha-256, RSA-4096` whereas the test cert is `sha1, RSA-2048`. Perhaps that is a cause of the difference?

**Comment by amcappelli** [comment]
Ok I think I figured it out - I found that when I ran chef-client as my user, Windows opened a prompt like in https://social.technet.microsoft.com/Forums/ie/en-US/aac8cfcd-6bd2-423f-895b-a6612459eb16/importcertificate-without-confirmation-ignore-security-warnings?forum=winserverpowershell. When I clicked Yes, chef-client finished successfully. When chef-client is run via a scheduled task as system, it fails because it isn't able to send that prompt. Looks like the old code in 4.2.5 uses the same strategy that that article suggests as a workaround which is why it works. Is there any chance of getting this change to use the ruby gem reverted (or a real fix in place) quickly? There are other changes in 4.3.0+ that we'd love to use so we don't particularly want to pin to < 4.3.0, but this currently breaks our entire provisioning system for windows.

Also, the test cert `GlobalSignRootCA` is already included in windows by default so it is not the best test. I confirmed that after deleting it from the root store and re-running chef-client to add it, I got the same error as with our internal CA cert that I'm trying to add.
